### DISCUSSION TITLE
question about multiple recasts?

### DISCUSSION THREAD

**Discussion by oakmac** [farcasterxyz/protocol#36] [General]
> A Recast expresses an intent to share another cast. An account can recast any cast, including its own, but **may not recast the same cast multiple times**.

I was reading through the Farcaster protocol and the bold text stuck out to me as an odd constraint.

I am curious what the reasoning is for not allowing multiple recasts at different times?

**Comment by pfletcherhill** [comment]
Just so I understand your question...are you asking about the case where I want to recast message A today and then recast the same message A again in a year?

**Comment by oakmac** [comment]
> the case where I want to recast message A today and then recast the same message A again in a year?

Yes

<hr>

I thought of the following use case:

- Paul the politician is working to create a new law. The law is passed and Paul casts a message about it "Today we passed law XYZ..."
- John the journalist writes a story about this new law and re-casts Paul's message, essentially sharing the news about the law to his audience.
- A few years pass and the law becomes unpopular / controversial / polarizing, etc. Paul is at the forefront of this situation.
- John writes a new story about the controversy and wishes to re-cast the same cast from a few years ago (to share the historical context with his audience), but is unable to due to the protocol design.

John can always prove that Paul's original cast came from Paul, but that feels different to me than a re-cast to Paul's audience. Essentially, **when** the re-cast occurs is an important aspect of the re-cast (in addition to the re-cast message content).

**Comment by varunsrin** [comment]
@oakmac to clarify, you can recast the same cast later but it will overwrite the previous recast.

**Comment by oakmac** [comment]
> you can recast the same cast later but it will overwrite the previous recast

I'm not sure what "overwrite the previous recast" means in the context of a conversation between people. If I say something to you in February, and then say the same thing to you in October, it doesn't mean that I didn't also say it to you in February.

My $0.02 is that the value in a cast comes from 1) the content and 2) the time at which that content was sent. Just like conversations in real life.

**Comment by varunsrin** [comment]
If you "like" something on most social networks in Feb, unlike it in March and like it again in April you don't see the full history of that. The rationale is that most users just care about the current state and eliminating older state by using last-write-wins rules saves storage space, which is particularly important on decentralized networks like Farcaster. 

Farcaster's POV on recasts is that they should be treated similar to likes and the intent is not to have a conversation, but to simply "reshare" something in your timeline at the current point in time. Older "reshares" becomes irrelevant when the same post is reshared in the future. Keeping a full history of every reshare is unnecessary from a UX perspective and expensive from a storage perspective. This is also similar to how retweets are implemented on Twitter btw, I don't believe they show up multiple times in your timeline, likely for the same reasons. 

Finally, if you really really want a the same cast to show up multiple times you can embed the cast as a link (instead of using the recast type) which will achieve want you want it to do.
